ntbs-service/Jobs/DataQualityAlertsJob.cs

```diff
@@ -23,229 +25,88 @@ public class DataQualityAlertsJob
+            _dataQualityRepository.GetNotificationsEligibleForDqDotVotAlertsCountAsync,
+            _dataQualityRepository.GetMultipleNotificationsEligibleForDqDotVotAlertsAsync
+        );
+
+        private async Task CreateTreatmentOutcome12MonthAlertsInBulk() => await CreateAlertsInBulkAsync<DataQualityTreatmentOutcome12>(
+            _dataQualityRepository.GetNotificationsEligibleForDqTreatmentOutcome12AlertsCountAsync,
+            _dataQualityRepository.GetMultipleNotificationsEligibleForDqTreatmentOutcome12AlertsAsync
+        );
+        
+        private async Task CreateTreatmentOutcome24MonthAlertsInBulk() => await CreateAlertsInBulkAsync<DataQualityTreatmentOutcome24>(
+            _dataQualityRepository.GetNotificationsEligibleForDqTreatmentOutcome24AlertsCountAsync,
+            _dataQualityRepository.GetMultipleNotificationsEligibleForDqTreatmentOutcome24AlertsAsync
+        );
+
+        private async Task CreateTreatmentOutcome36MonthAlertsInBulk()  => await CreateAlertsInBulkAsync<DataQualityTreatmentOutcome36>(
+            _dataQualityRepository.GetNotificationsEligibleForDqTreatmentOutcome36AlertsCountAsync,
+            _dataQualityRepository.GetMultipleNotificationsEligibleForDqTreatmentOutcome36AlertsAsync
+        );
+        
+        private async Task CreateAlertsInBulkAsync<T>(
+            GetNotificationsEligibleForDqAlertsCount getCount,
+            GetMultipleNotificationsEligibleForDataQualityDraftAlerts getNotifications) where T : Alert
         {
-            var notificationsForAlertsCount = await _dataQualityRepository
-                .GetNotificationsEligibleForDqTreatmentOutcome36AlertsCount();
-            
-            var x = 0;
-            while (x < notificationsForAlertsCount)
+            var notificationsForAlertsCount = await getCount();
+            var offset = 0;
+            while (offset < notificationsForAlertsCount)
             {
-                var alertsToAdd = new List<Alert>();
-                var notificationsForAlerts =
-                    await _dataQualityRepository.GetMultipleNotificationsEligibleForDqTreatmentOutcome36Alerts(CountPerBatch, x);
-                foreach (var notification in notificationsForAlerts)
-                {
-                    
-                    var alert = new DataQualityTreatmentOutcome36 {NotificationId = notification.NotificationId};
-                    var populatedAlert = await _alertService.CheckIfAlertExistsAndPopulateIfItDoesNot(alert, notification);
-
-                    if (populatedAlert != null)
-                    {
-                        alertsToAdd.Add(populatedAlert);   
-                    }
-                }
-
-                await _alertService.AddRangeAlerts(alertsToAdd);
-                x += 500;
+                var notificationsForAlerts = await getNotifications(CountPerBatch, offset);
+                
+                List<Alert> dqAlerts = notificationsForAlerts
```

Comment:
Could this not be `List<T>`? or does the `AddAlertsRangeAsync` complain when it is?